### PR TITLE
Fix lcmtypes path inconsistencies

### LINKED ISSUE #64: move lcmtypes into their most relevant directories and exploit the new lcmtypes.cmake

### LINKED ISSUE #1062: Drake lcmtype c++ headers have wrong includes
Drake's c++ headers get installed to `build/include/lcmtypes/drake/`, so you would typically want to include on with:

```
#include "lcmtypes/drake/lcmt_qp_controller_input.hpp"
```

but the headers themselves reference other lcmtypes without the `lcmtypes` prefix. For example, the beginning of `lcmt_qp_controller_input.hpp` looks like:

```
#include "drake/lcmt_zmp_data.hpp"
#include "drake/lcmt_support_data.hpp"
#include "drake/lcmt_body_motion_data.hpp"
#include "drake/lcmt_body_wrench_data.hpp"
#include "drake/lcmt_whole_body_data.hpp"
#include "drake/lcmt_joint_pd_override.hpp"
```

This makes it impossible to include any Drake c++ headers without playing around with the includes path in cmake.

### PR DESCRIPTION
**Pull request by rdeits**
See https://github.com/RobotLocomotion/cmake/pull/17

Now all lcmtypes refer to other lcmtypes using their full path from the `include/` folder. That means that it's easy to include any drake lcmtypes without manipulating the includes path at all. The only possible disadvantage is that you need to specify the `lcmtypes` prefix when you include an lcmtype header, but, as I said here: https://github.com/RobotLocomotion/cmake/pull/17 I think that's actually a good thing. 

Note that this switches the cmake external to point to my fork. That's intentional and temporary so that the build servers can work on this before the cmake pull request gets merged. If we're happy with this, then we can merge https://github.com/RobotLocomotion/cmake/pull/17 and then I'll update this branch to point back to cmake master. 

Fixes #1062 
Fixes #64

### CODE REVIEW

**Comment by rdeits** [comment]
I guess it's about time that I took care of issue #64 :wink:

File: drake/lcmtypes/CMakeLists.txt [hunk 1]
```diff
@@ -1,31 +0,0 @@
-
```
Comment (rdeits) [inline drake/lcmtypes/CMakeLists.txt]: this file wasn't actually being used by anything

File: CMakeLists.txt [hunk 2]
```diff
@@ -103,8 +103,8 @@ set(bertini_GIT_REPOSITORY https://github.com/RobotLocomotion/bertini.git)
 set(bertini_GIT_TAG 3ae3e3ad3534acb4b6f7a4c3c22728b493beaa80)
 set(bullet_GIT_REPOSITORY https://github.com/RobotLocomotion/bullet-pod.git)
 set(bullet_GIT_TAG 6db7ae5e29f8b35ffcea384c53caa4bddcc6f986)
-set(cmake_GIT_REPOSITORY https://github.com/RobotLocomotion/cmake.git)
-set(cmake_GIT_TAG c48d1e461d8f0a8d67ebba3e4acb19d96dd5a25b)
+set(cmake_GIT_REPOSITORY https://github.com/rdeits/cmake.git)
+set(cmake_GIT_TAG f48840614d7d80a290ba0ef63ee1166b32ebcfac)
```
Comment (RussTedrake) [inline CMakeLists.txt]: i've merged the other.  can we keep it pointing to RobotLocomotion please?

**Comment by RussTedrake** [comment]
otherwise looks good.

**Comment by rdeits** [comment]
ok, done

File: drake/solvers/CMakeLists.txt [hunk 3]
```diff
@@ -27,8 +27,10 @@ endif()
 
 pods_find_pkg_config(snopt_c)
 
-add_mex(lqrmex lqrmex.cpp)
-target_link_libraries(lqrmex drakeUtil)
+if(MATLAB_FOUND)
+	add_mex(lqrmex lqrmex.cpp)
+	target_link_libraries(lqrmex drakeUtil)
```
Comment (tkoolen) [inline drake/solvers/CMakeLists.txt]: two spaces instead of tab

**Comment by tkoolen** [comment]
Thank you!

**Comment by rdeits** [comment]
@tkoolen fixed (actually, that file was using a mix of tabs, 2 spaces, and 4 spaces. It should be consistently 2 spaces now)

**Pull request closed without merging**